### PR TITLE
AK/MIN-191 ControllerAPI

### PR DESCRIPTION
**Pull request by kurotychb**
**Description:**

Implemented ControllerAPI and adjusted others pallet to use it 

**Changes in storage or API:**

Describe storage structure changes or API changes. Make sure to update the [API docs](https://minterestfinance.atlassian.net/wiki/spaces/MINTEREST/pages/134807557/Minterest+API).

**Checklist**

- [ ] Code is covered with tests
- [ ] Code is covered with benchmarks
- [ ] Fresh benchmark results included

### CODE REVIEW

File: pallets/controller/src/lib.rs [hunk 1]
```diff
@@ -931,7 +749,190 @@ impl<T: Config> Pallet<T> {
 }
 
 impl<T: Config> ControllerAPI<T::AccountId> for Pallet<T> {
```
Comment (lut1k) [inline pallets/controller/src/lib.rs:751]: Different impl member order from the trait.

File: test-helper/src/lib.rs [hunk 2]
```diff
@@ -198,21 +198,22 @@ macro_rules! mock_impl_dex_config {
...
 			type ProtocolWeightInfo = ();
+			type ControllerAPI = $controller_api;
 		}
 	};
 }
 
 #[macro_export]
 macro_rules! mock_impl_risk_manager_config {
-	($target:ty, $acc:ident) => {
+	($target:ty, $acc:ident, $controller_api:ty) => {
```
Comment (lut1k) [inline test-helper/src/lib.rs:216]: I think there is no need to take the pallet controller module from outside:
```
#[macro_export]
macro_rules! mock_impl_risk_manager_config {
	($target:ty, $acc:ident) => {
		parameter_types! {
			pub const RiskManagerPriority: TransactionPriority = TransactionPriority::max_value();
		}

		impl risk_manager::Config for $target {
			type Event = Event;
			type UnsignedPriority = RiskManagerPriority;
			type LiquidationPoolsManager = liquidation_pools::Module<$target>;
			type LiquidityPoolsManager = liquidity_pools::Module<$target>;
			type RiskManagerUpdateOrigin = EnsureSignedBy<$acc, AccountId>;
			type RiskManagerWeightInfo = ();
			type ControllerAPI = controller::Module<$target>;
		}
	};
}
```

File: test-helper/src/lib.rs [hunk 3]
```diff
@@ -198,21 +198,22 @@ macro_rules! mock_impl_dex_config {
 
 #[macro_export]
 macro_rules! mock_impl_minterest_protocol_config {
-	($target:ty) => {
+	($target:ty, $controller_api:ty) => {
```
Comment (lut1k) [inline test-helper/src/lib.rs:201]: I think there is no need to take the pallet controller module from outside:
```
#[macro_export]
macro_rules! mock_impl_minterest_protocol_config {
	($target:ty) => {
		impl minterest_protocol::Config for $target {
			type Event = Event;
			type Borrowing = liquidity_pools::Module<$target>;
			type ManagerLiquidationPools = liquidation_pools::Module<$target>;
			type ManagerLiquidityPools = liquidity_pools::Module<$target>;
			type WhitelistMembers = WhitelistMembers;
			type ProtocolWeightInfo = ();
			type ControllerAPI = controller::Module<$target>;
		}
	};
}
```

File: pallets/risk-manager/src/mock.rs [hunk 4]
```diff
@@ -64,8 +64,8 @@ mock_impl_liquidation_pools_config!(Test);
 mock_impl_controller_config!(Test, ZeroAdmin);
 mock_impl_minterest_model_config!(Test, ZeroAdmin);
 mock_impl_dex_config!(Test);
-mock_impl_minterest_protocol_config!(Test);
-mock_impl_risk_manager_config!(Test, ZeroAdmin);
+mock_impl_minterest_protocol_config!(Test, Controller);
+mock_impl_risk_manager_config!(Test, ZeroAdmin, Controller);
```
Comment (lut1k) [inline pallets/risk-manager/src/mock.rs]: Should be removed according to previous comments.

**Review by lut1k — APPROVED** [review]

**Review by dev-vadim — DISMISSED** [review]

**Review by Apollo50 — APPROVED** [review]

**Review by lut1k — APPROVED** [review]

**Review by Apollo50 — APPROVED** [review]

**Pull request merged**